master/static/srv/get_monthly_aggregated_allocation.sql

```diff
@@ -0,0 +1,25 @@
+WITH const AS (
+    SELECT
+        daterange($1 :: date, $2 :: date, '[]') AS period
+)
+SELECT
+    to_char(resource_aggregates.date :: date, 'YYYY-MM') AS date,
+    -- These values must be kept in sync with the protobuf type ResourceAllocationAggregationType.
+    CASE
+        WHEN resource_aggregates.aggregation_type = 'total' THEN 'RESOURCE_ALLOCATION_AGGREGATION_TYPE_TOTAL'
+        WHEN resource_aggregates.aggregation_type = 'user' THEN 'RESOURCE_ALLOCATION_AGGREGATION_TYPE_USER'
+        WHEN resource_aggregates.aggregation_type = 'label' THEN 'RESOURCE_ALLOCATION_AGGREGATION_TYPE_LABEL'
+        ELSE 'RESOURCE_ALLOCATION_AGGREGATION_TYPE_UNSPECIFIED'
+    END AS aggregation_type,
+    resource_aggregates.aggregation_key,
+    sum(resource_aggregates.seconds) AS seconds
+FROM
+    resource_aggregates,
+    const
+WHERE
+    -- `@>` determines whether the range contains the time.
+    const.period @> resource_aggregates.date
```

Comment:
did we consider a btree index on resource_aggregates.date? I think i remember the conversation somewhere.. if there are N users, M labels, this is what, N + M rows per date? If you query for a month maybe ~10^3 rows, after a year maybe ~10^4 rows in the table. By dumb math, it makes me think an index could speed up by a factor of 10 after this exists a while?

Reply:
Sounds good, I'll add that and land.

Reply:
Oh, there's already a unique constraint, which [automatically adds an index](https://www.postgresql.org/docs/10/indexes-unique.html).